dpytools/embeds.py

```diff
@@ -118,3 +119,141 @@ def add_fields(self, inline=True, **kwargs):
+        Number of field in each embed. Must be greater then 0. Defaults to 25
+    color: Optional[:class:`discord.Color`, :class: `int`]
+        The color of the disord embed. Default is Black
+    """
+
+    def __init__(self, **options):
+        self.dicts = options.pop('dicts')
+        self.title = options.pop('title', '')
+        self.description = options.pop('description', '')
+        self.footer = options.pop('footer', '')
+        self.author = options.pop('author', '')
+        self.inline = options.pop('inline', True)
+        self.colour = options.pop('colour', 0)
+        self.size = options.pop('size', 25)
+        
+        self.field_limit = 25
+        self.char_limit = 6000
+        self.clear()
+        self._add_embed(self.dicts)
+
+    def clear(self):
+        """Clears the paginator to have no pages."""
+        self._pages = []
+
+    def _check_embed(self, embed: discord.Embed, *chars: str):
+        """
+        Check if the emebed is too big to be sent on discord
+
+        Args:
+            embed (discord.Embed): The embed to check
+
+        Returns:
+            bool: Will return True if the emebed isn't too large
+        """
+        check = (
+            len(embed) + sum(len(char) for char in chars if char) < self.char_limit
+            and len(embed.fields) < self.field_limit
+        )
+        return check
+
+    def _new_page(self):
+        """
+        Create a new page
+
+        Args:
+            title (str): The title of the new page
+
+        Returns:
+            discord.Emebed: Returns an embed with the title and color set
+        """
+        return discord.Embed(title=self.title, description=self.description, timestamp=datetime.now(), color=self.colour)
+
+    def _add_page(self, page: discord.Embed):
+        """
+        Add a page to the paginator
+
+        Args:
+            page (discord.Embed): The page to add
+        """
+        page.set_footer(text=self.footer)
+        page.set_author(name=self.author)
+        self._pages.append(page)
+
+    def _chunks(self, dicts:dict):
+        """Yield successive chunks of num size from dicts
+
+        Args:
+            dicts (dict): Dictionary containing name and value
+
+        Yields:
+            tuple_list (list): list chunks of size num containing name and value
+        """        
+
+        num = self.size
+        if not dicts:
+            raise ValueError("Dictionary containing name and value can't be empty!")
+
+        tuple_list = [(name, value) for name, value in dicts.items()]
+
+        if num < 1:
+            raise ValueError("Number of Embed fields can't be zero")
+
+        for i in range(0, len(tuple_list), num):
+            yield tuple_list[i:i+num]
+    
+    def _add_embed(self, dicts):
+        """Add embeds to pages
+
+        Args:
+            dicts :- Name, Value dictionary to generate embed from
+
+        Returns:
+            None
+
+        """        
+        for d in self._chunks(dicts):
```

Comment:
Same as before, variable names in this loop are too ambiguous. Try to name them something that actually tells what they are or what they do